packages/utils/telemetry-utils/src/sampledTelemetryHelper.ts

```diff
@@ -98,23 +141,33 @@ export class SampledTelemetryHelper implements IDisposable {
 	/**
 	 * Executes the specified code and keeps track of execution time statistics.
 	 * If it's been called enough times (the sampleThreshold for the class) then it generates a log message with the necessary information.
+	 * @remarks The user MUST pass in exact same SET of custom data properties and the set must be IMMUABLE during the life cycle of the logger instance.
 	 *
 	 * @param codeToMeasure - The code to be executed and measured.
 	 * @param bucket - A key to track executions of the code block separately.
 	 * Each different value of this parameter has a separate set of executions and metrics tracked by the class.
 	 * If no such distinction needs to be made, do not provide a value.
 	 * @returns Whatever the passed-in code block returns.
 	 */
-	public measure<T>(codeToMeasure: () => T, bucket: string = ""): T {
+	public measure<T>(
+		codeToMeasure: () => void | (T & ICustomData<TCustomMetrics>),
```

Comment:
The return type of `codeToMeasure` has gotten esoteric (so as the return type of `measure()`) compared to the initial type of `T`.